index.js

```diff
@@ -282,7 +282,7 @@ is.arraylike = function (value) {
   return !!value && !is.bool(value)
     && owns.call(value, 'length')
     && isFinite(value.length)
-    && is.number(value.length)
+    && is.integer(value.length)
```

Comment:
That's a fair point. However, while people might be relying on non-integers being valid, I don't think anyone's expecting `{ length: NaN }` or `{ length: Infinity }` to work :-)